src/beta/resources/referral.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-dupe-class-members,no-unused-vars,class-methods-use-this */
+import superagent from 'superagent';
+
+import T from 'proptypes';
+import base from '../../resources/base';
+import { propTypes as userPropTypes } from '../../resources/user';
+import RequestError from '../../errors/request';
+import API from '../../easypost';
+
+export const propTypes = Object.assign({}, userPropTypes, {
+  name: T.string.isRequired, // mark as required
+  email: T.string.isRequired, // mark as required
+  phone: T.string.isRequired, // handle discrepancy between phone (create Referral) and phone_number (all User functions, retrieve Referral)
+});
+
+export default (api) =>
+  class Referral extends base(api) {
+    static _name = 'Referral';
+
+    static _url = 'referral_customers';
+
+    static key = 'user';
+
+    static propTypes = propTypes;
+
+    static #stripeCreditCardUrl = 'https://api.stripe.com/v1/tokens';
```

Comment:
My understanding is that the "private function/vars" addition of `#` was only added in Node 12+ and we support Node 10. We'll need to look at a pollyfill to allow this to get backported to Node 10. It may not have failed here because we're simply replaying cassettes and you didn't record them on Node 10 (could very well fail at runtime): https://stackoverflow.com/a/56183753/6064135

This is true for all `#` usage in this PR